test/common/auth/token_storage/token_storage.test.ts

```diff
@@ -0,0 +1,110 @@
+import { describe, it, expect, test, vi, afterEach, beforeEach } from "vitest";
+import type { AuthData } from "home-assistant-js-websocket";
+
+describe("token_storage", () => {
+  beforeEach(() => {
+    vi.stubGlobal(
+      "window.__tokenCache",
+      (window.__tokenCache = {
+        tokens: undefined,
+        writeEnabled: undefined,
+      })
+    );
+  });
+
+  afterEach(() => {
+    vi.resetModules();
+  });
+
+  test("initialize tokenCache", async () => {
+    vi.stubGlobal(
+      "window.__tokenCache",
+      (window.__tokenCache = undefined as any)
+    );
+
+    await import("../../../../src/common/auth/token_storage");
+
+    expect(window.__tokenCache).toEqual({
+      tokens: undefined,
+      writeEnabled: undefined,
+    });
+  });
+
+  test("should load tokens", async () => {
+    const tokens: AuthData = {
+      access_token: "test",
+      expires: 1800,
+      expires_in: 1800,
+      hassUrl: "http://localhost",
+      refresh_token: "refresh",
+      clientId: "client",
+    };
+
+    window.localStorage = {
+      getItem: vi.fn(() => JSON.stringify(tokens)),
+    } as unknown as Storage;
```

Comment:
this doesnt quard us from using a wrong keyname when saving of retrieving, can we add that?